File: python/example_code/polly/SynthesizeSpeech.py
# snippet-comment:[These are tags for the AWS doc team's sample catalog. Do not remove.]
# snippet-sourcedescription:[SynthesizeSpeech.py demonstrates how to synthesize speech with shorter texts for near-real-time processing.]
# snippet-service:[polly]
# snippet-keyword:[Amazon Polly]
# snippet-keyword:[Python]
# snippet-sourcesyntax:[python]
# snippet-sourcesyntax:[python]
# snippet-sourcedate:[2019-03-13]
# snippet-sourceauthor:[AWS]

# Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
#
# This file is licensed under the Apache License, Version 2.0 (the "License").
# You may not use this file except in compliance with the License. A copy of the
# License is located at
#
# http://aws.amazon.com/apache2.0/
#
# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
# OF ANY KIND, either express or implied. See the License for the specific
# language governing permissions and limitations under the License.

# snippet-start:[polly.python.SynthesizeSpeech.complete]
import boto3

# Synthesize the sample text, saving it in an MP3 audio file
polly_client = boto3.client('polly')
response = polly_client.synthesize_speech(VoiceId='Joanna',
                                          OutputFormat='mp3',
                                          Text='This is sample text to synthesize.')
with open('speech.mp3', 'w') as file:
    file.write(response['AudioStream'].read())
# snippet-end:[polly.python.SynthesizeSpeech.complete]

File: php/example_code/polly/SynthesizeSpeech.php
<?php
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

// snippet-start:[polly.php.synthesize_speech.complete]
// snippet-start:[polly.php.synthesize_speech.import]

require 'vendor/autoload.php';

use Aws\Exception\AwsException;

// snippet-end:[polly.php.synthesize_speech.import]

/**
 * This code expects that you have AWS credentials set up per:
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials.html
 */
// snippet-start:[polly.php.synthesize_speech.main]
// Create a PollyClient
$client = new Aws\Polly\PollyClient([
    'profile' => 'default',
    'version' => '2016-06-10',
    'region' => 'us-east-2'
]);

$text = 'This is a sample text to be synthesized.';
$format = 'json'; //json|mp3|ogg_vorbis|pcm
$voice = 'Joanna';

try {
    $result = $client->synthesizeSpeech([
        'Text' => $text,
        'OutputFormat' => $format,
        'VoiceId' => $voice,
    ]);
    var_dump($result);
} catch (AwsException $e) {
    // output error message if fails
    echo $e->getMessage();
    echo "\n";
}

// snippet-end:[polly.php.synthesize_speech.main]
// snippet-end:[polly.php.synthesize_speech.complete]
